compiler/rustc_infer/src/infer/lexical_region_resolve/mod.rs

```diff
@@ -247,15 +214,18 @@ impl<'cx, 'tcx> LexicalResolver<'cx, 'tcx> {
                                 true
                             }
                             VarValue::Value(cur_region) => {
-                                let lub = match self.lub_empty(cur_region) {
-                                    Ok(r) => r,
-                                    // If the empty and placeholder regions are in the same universe,
-                                    // then the LUB is the Placeholder region (which is the cur_region).
-                                    // If they are not in the same universe, the LUB is the Static lifetime.
-                                    Err(placeholder) if a_universe == placeholder.universe => {
-                                        cur_region
+                                let lub = match *cur_region {
```

Comment:
instead of using `lub` and then checking it to be equal to `cur_region`, merge the match and then `if`